File: core/src/com/desitum/library/game/G2DSprite.java
package com.desitum.library.game;

import com.badlogic.gdx.graphics.g2d.NinePatch;
import com.badlogic.gdx.scenes.scene2d.utils.NinePatchDrawable;
import com.desitum.library.view.TouchEvent;

/**
 * Created by kodyvanry on 5/2/17.
 */

public abstract class G2DSprite extends com.badlogic.gdx.graphics.g2d.Sprite implements Comparable<G2DSprite> {

    private int mLayer = 0;

    public G2DSprite() {
        super();
    }

    public void setLayer(int mLayer) {
        this.mLayer = mLayer;
    }

    public int getLayer() {
        return mLayer;
    }

    @Override
    public int compareTo(G2DSprite o) {
        return mLayer - o.mLayer;
    }

    public abstract void update(float delta);

    public boolean onTouchDown(TouchEvent touchEvent) {
        return false;
    }

    public boolean onTouchMoved(TouchEvent touchEvent) {
        return false;
    }

    public boolean onTouchUp(TouchEvent touchEvent) {
        return false;
    }
}

File: core/src/com/desitum/library/game/G2DSprite.kt
package com.desitum.library.game

import com.desitum.library.view.TouchEvent

/**
 * Created by kodyvanry on 5/2/17.
 */

abstract class G2DSprite : com.badlogic.gdx.graphics.g2d.Sprite(), Comparable<G2DSprite> {

    var z = 0

    override fun compareTo(other: G2DSprite): Int {
        return z - other.z
    }

    abstract fun update(delta: Float)

    fun onTouchDown(touchEvent: TouchEvent): Boolean {
        return false
    }

    fun onTouchMoved(touchEvent: TouchEvent): Boolean {
        return false
    }

    fun onTouchUp(touchEvent: TouchEvent): Boolean {
        return false
    }
}
